test/HttpApiClientTest.ts

```diff
@@ -0,0 +1,116 @@
+import {HttpApiClient, HttpClient} from "../src/httpClient";
+
+describe("HttpApiClient", () => {
+    let httpApiClient: HttpApiClient;
+
+    it("should return a valid baseUrl after construction", () => {
+
+        let defaultBaseUrl: string;
+
+        let httpClient = new HttpClient();
+        httpClient.configure((cfg) => {
+            defaultBaseUrl = cfg.options.baseUrl;
+        });
+
+        httpApiClient = new HttpApiClient();
+        httpApiClient.configure((cfg) => {
+            expect(cfg.options.baseUrl).toEqual(defaultBaseUrl);
+        });
+
+        httpApiClient = new HttpApiClient();
+        expect(httpApiClient.baseUrl).toBeUndefined();
+
+        httpApiClient = new HttpApiClient("https://myapi.crazy-factory.com");
+        expect(httpApiClient.baseUrl).toEqual("https://myapi.crazy-factory.com");
+
+    });
+
+    it("should throw errors for incorrect parameters", () => {
```

Comment:
maybe rephrase that to "incorrect method" and also test for undefined/null here.